### PR TITLE
Lehins ts/split libs

### PR DESCRIPTION
**Pull request by TimSheard**
Split small-steps into three parts
  small-steps (Control.State.Transition, Control.State.Transition.Extended, ...)
  cardano-data  (Data.Coders, Data.MemoBytes, Data.BiMap, Data.Sharing ...)
  set-algebra (Control.Iterate.BaseTypes, Control.Iterate.SetAlgebra, ...)

This makes it possible for set-algebra to depend on cardano-ledger-core, this supports Set Algebra expression over types that know about Core types.

The vast majority of the 42 files changed, are cabal files that have added depends on for cardano-data and set-algebra

### CODE REVIEW

File: libs/cardano-data/default.nix [hunk 1]
```diff
@@ -0,0 +1,4 @@
+let
+  pkgs = import ../../../default.nix {};
+in
+  pkgs.libs.small-steps
```
Comment (lehins) [inline libs/cardano-data/default.nix]: Looks like we missed some
```suggestion
  pkgs.libs.cardano-data
```

File: libs/set-algebra/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Revision history for small-steps
```
Comment (lehins) [inline libs/set-algebra/CHANGELOG.md]:
```suggestion
# Revision history for set-algebra
```

File: libs/set-algebra/default.nix [hunk 3]
```diff
@@ -0,0 +1,4 @@
+let
+  pkgs = import ../../../default.nix {};
+in
+  pkgs.libs.small-steps
```
Comment (lehins) [inline libs/set-algebra/default.nix]:
```suggestion
  pkgs.libs.set-algebra
```

File: libs/cardano-data/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Revision history for small-steps
```
Comment (lehins) [inline libs/cardano-data/CHANGELOG.md]:
```suggestion
# Revision history for cardano-data
```

**Review by lehins — CHANGES_REQUESTED** [review]
We forgot couple of renames and also I forgot to mention that we need to add these two settings intro the top level `cabal.project` in order to turn warnings into errors:

```
package set-algebra
  ghc-options: -Werror

package cardano-data
  ghc-options: -Werror
```


And of course it needs a rebase on master

File: libs/cardano-data/default.nix (hunk 1 above)
Comment (TimSheard) [reply libs/cardano-data/default.nix]: done

File: libs/set-algebra/CHANGELOG.md (hunk 2 above)
Comment (TimSheard) [reply libs/set-algebra/CHANGELOG.md]: done

File: libs/set-algebra/default.nix (hunk 3 above)
Comment (TimSheard) [reply libs/set-algebra/default.nix]: done

File: libs/cardano-data/CHANGELOG.md (hunk 4 above)
Comment (TimSheard) [reply libs/cardano-data/CHANGELOG.md]: done

**Review by JaredCorduan — APPROVED** [review]
thanks @TimSheard !

**Review by lehins — APPROVED** [review]

**Pull request merged**